deep-learning/_posts/2018-10-18-residual-iterative.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: review
+title:  "Residual Connections Encourage Iterative Inference"
+tags:   deep-learning layers
+author: Carl Lemaire
+pdf:    https://arxiv.org/pdf/1710.04773.pdf
+cite:
+  authors: "Stanisław Jastrzebski, Devansh Arpit, Nicolas Ballas, Vikas Verma, Tong Che, Yoshua Bengio"
+  title:   "Residual Connections Encourage Iterative Inference"
+  venue:   "ICLR 2018"
+---
+
+In this paper, the authors examine the concept of **residual connections** through the lens of incremental/iterative inference.
+
+# ResBlocks: "Refinement" blocks?
+
+It makes a lot of sense to view resblocks as "refinement" blocks. Let me explain. The typical way of illustrating resblocks is the following. (For simplicity, let's ignore the spatial dimensions, and use only one layer per block):
+
+![](/deep-learning/images/residual-iterative/dessin1.png)
+
+In my opinion, this is misleading. The residual connections seem to be a "bonus", just something you add to any good old feedforward network. Here is an illustration that I think is more representative:
+
+![](/deep-learning/images/residual-iterative/dessin2.png)
+
+Same network, different interpretation. Here, the residual connections are drawn as straight lines, highlighting the continuity of the features throughout the layers. Now, the residual branch becomes the "main branch", and the secondary branch computes a "delta" that "refines" the existing signal. This illustration also highlights the fact that a ResBlock do not (generally) change the number of features (or change the spatial dimensions) of the main signal.
+
+# Iterative refinement in ResNets
+
+The authors provide 3 interesting contributions:
+
+1. **They find that residual blocks are encouraged to produce deltas that are in the direction of the negative loss gradient**. They empirically confirm this by comparing (with cosine distance) the delta produced by a resblock and the gradient of the loss relative to the hidden features prior to the block.
+
+    ![](/deep-learning/images/residual-iterative/fig4.png)
+
+2. **They observe that resnets can perform both hierarchical reprensentation learning and iterative feature refinement**. They find that lower layers focus on representation learning, and upper layers on refinement. Since the upper layers are only refining, they can be removed with minimal effect on the accuracy.
```

Comment:
reprensentation -> representation

Reply:
The idea that upper blocks are only refining is not a contribution from this paper, right? So I do not see how this relates to first the statement.

Reply:
They experimentally show that upper blocks are only refining

Reply:
OK. Thanks for answering this Carl.